### PR TITLE
use new URL in `subscribeSafariPromptPermission`

### PR DESCRIPTION
**Pull request by fedterzi**
# Description
## 1 Line Summary
use new URL with `appId` in `subscribeSafariPromptPermission`

## Details
Due to changes in our API, the `appId` is now required in the URL used by `subscribeSafariPromptPermission`. In instances where the new URL is not usable, we fallback to the legacy URL with an additional request. This ensures compatibility with both the updated and legacy API endpoints.

# Systems Affected
   - [x] WebSDK
   - [x] Backend
   - [ ] Dashboard

# Validation
### Checklist
   - [ ] All the automated tests pass or I explained why that is not possible
   - [ ] I have personally tested this on my machine or explained why that is not possible
   - [ ] I have included test coverage for these changes or explained why they are not needed

**Programming Checklist**
Interfaces:
   - [ ] Don't use default export
   - [ ] New interfaces are in model files

Functions:
   - [ ] Don't use default export
   - [ ] All function signatures have return types
   - [ ] Helpers should not access any data but rather be given the data to operate on.

Typescript:
   - [ ] No Typescript warnings
   - [ ] Avoid silencing null/undefined warnings with the exclamation point

Other:
   - [ ] Iteration: refrain from using `elem of array` syntax. Prefer `forEach` or use `map`
   - [ ] Avoid using global OneSignal accessor for `context` if possible. Instead, we can pass it to function/constructor so that we don't call `OneSignal.context`

### Checklist
   - [ ] I have included screenshots/recordings of the intended results or explained why they are not needed

---

## Related Tickets

---

- - -
This change is [Reviewable](https://reviewable.io/reviews/OneSignal/OneSignal-Website-SDK/1138)

### CODE REVIEW

File: src/shared/managers/SubscriptionManager.ts [hunk 1]
```diff
@@ -372,20 +372,34 @@ export class SubscriptionManager {
...
+          this.config.safariWebId,
+          { app_id: this.config.appId },
+          (response) => {
+            if ((response as any).deviceToken) {
+              resolve((response as any).deviceToken.toLowerCase());
+            } else {
+              // If there's no token and the new URL was used,
+              // retry with the legacy URL
+              if (url.includes('/safari/apps/')) {
+                requestPermission(
```
Comment (jkasten2) [inline src/shared/managers/SubscriptionManager.ts:388]: This recursive call makes the code really hard to follow. Could we simply call `await requestPermission(...)`, check if we get a value, then call `await requestPermission(...)` again with the fallback URL?

**Review by jkasten2 — CHANGES_REQUESTED** [review]

File: src/shared/managers/SubscriptionManager.ts (hunk 1 above)
Comment (fedterzi) [reply src/shared/managers/SubscriptionManager.ts]: Done.

**Review by jkasten2 — APPROVED** [review]

**Review by jkasten2 — CHANGES_REQUESTED** [review]
I tested this and unfortunately the fallback call to `window.safari.pushNotification.requestPermission` doesn't work with Safari. The browser throws the following error "Push notification prompting can only be done from a user gesture.". It seems even though there is an `await` the browser doesn't consider it the same event.

**Review by jkasten2 — APPROVED** [review]
Confirmed this works as expected in Safari 17.1 on macOS 13.6.1. First click does nothing, then the 2nd click results in the native Safari notification showing correctly. Once the Safari zips are regenerated server side then the first click will start working again.
